pkg/storage/sst.go

```diff
@@ -69,24 +68,50 @@ func CheckSSTConflicts(
 	defer sstIter.Close()
 	sstIter.SeekGE(start)
 
-	extOK, extErr := extIter.Valid()
+	// extIter is a prefix iterator; it is expected to skip keys that belong
+	// to different prefixes. Only iterate along the sst iterator, and re-seek
+	// extIter each time.
 	sstOK, sstErr := sstIter.Valid()
-	for extErr == nil && sstErr == nil && extOK && sstOK {
+	if sstOK {
+		extIter.SeekGE(MVCCKey{Key: sstIter.UnsafeKey().Key})
+	}
+	extOK, extErr := extIter.Valid()
+	for sstErr == nil && sstOK {
 		if err := ctx.Err(); err != nil {
 			return enginepb.MVCCStats{}, err
 		}
+		if !extOK {
```

Comment:
Am I reading this right that the fast-path if the target span is totally empty, where extIter would land above the sst and then sstIter would be exhausted, is gone now? Instead we'll (quickly) seek to every sst key now, right?

Reply:
Yeah, that fast path is gone. We don't have an alternative (maybe we could also spin up a non-prefix iter and check it once to see if the entire span is empty). The prefix iterator on its own doesn't help us here as it could skip keys.

Reply:
it seems like a non-prefix iter with an upper bound would be nice to check if the span is literally empty. We might even have one hanging out already in evalAddSSTable that would avoid needing to make a new one?

Reply:
Added a fast-path check, and re-ran benchmarks. Thanks!